ir/src/main/scala/org/scalajs/core/ir/Printers.scala

```diff
@@ -1052,4 +1052,601 @@ object Printers {
     def complete(): Unit = ()
   }
 
+  abstract class RawIndentationManager extends IndentationManager {
+    private var indentSize = 0
+    private val indentString = "  "
+
+    override def indent(): Unit = indentSize += 1
+    override def undent(): Unit = indentSize -= 1
+
+    protected def print(s: String): Unit = {
+      out.write(indentString * indentSize)
+      out.write(s)
+    }
+
+    protected def println(s: String): Unit =
+      print(s + "\n")
+
+    override protected def println(): Unit =
+      out.write("\n")
+  }
+
+  class IRASTPrinter(protected val out: Writer) extends RawIndentationManager {
+    /** Pretty-prints the IR's AST nodes.
+      */
```

Comment:
See https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md#scaladoc for the Scaladoc style to use.